spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1469,5 +1469,36 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+  test("unary negative integer overflow test") {
+    withTempDir { dir =>
+      val path = new Path(dir.toURI.toString, "int.parquet")
+      val df = Seq(Int.MaxValue, Int.MinValue).toDF("a")
+      df.write.mode("overwrite").parquet(path.toString)
+      spark.read.parquet(path.toString).createTempView("t")
+
+      // without ANSI mode
+      withSQLConf(
```

Comment:
I would say let's add scalar and dictionary tests

Reply:
@kazuyukitanimura can you please explain this a bit more?

Do you want me to create a new test to check non-overflow values?

Reply:
@kazuyukitanimura I have added a Scalar test.